### ISSUE TITLE
conformer fine-tuning with conformer_ctc_char.yaml

### ISSUE THREAD

**Issue by MelikaBahmanabadi** [NVIDIA/NeMo#4062]
Hi thanks for your great tutorial, 
I have an error `Interpolation key 'model.labels' not found` when I set up the training params model. I do the following steps:
1) I used `stt_en_conformer_ctc_small` as pre-trained model.
2) I load `conformer_ctc_char.yaml ` file.
3) I change the model.labels in this file manually to new language characters. 
4) I change the train test monifest path and lr.
5) fit the model.
I attach my code. so what is the problem? How can I fine-tune the conformer for another language in CTC char-based?

```
import nemo
import nemo.collections.asr as nemo_asr
asr_model = nemo_asr.models.EncDecCTCModelBPE.from_pretrained(model_name="stt_en_conformer_ctc_small")

from omegaconf import OmegaConf, open_dict
params = OmegaConf.load("conformer_ctc_char.yaml")
print(OmegaConf.to_yaml(params))

train_manifest = "./train.json"
test_manifest = "./test.json"
params.model.train_ds.manifest_filepath = train_manifest
params.model.validation_ds.manifest_filepath = test_manifest

params.model.spec_augment.rect_masks = 0

import copy
new_opt = copy.deepcopy(params.model.optim)
new_opt.lr = 0.001

print(params.model.optim)

print(asr_model.decoder.vocabulary)

asr_model.setup_optimization(optim_config=new_opt)

asr_model.setup_training_data(train_data_config=params['model']['train_ds'])

asr_model.setup_validation_data(val_data_config=params['model']['validation_ds'])

asr_model.encoder.freeze()

import pytorch_lightning as pl
from pytorch_lightning.callbacks.early_stopping import EarlyStopping

early_stop_callback = EarlyStopping(
   monitor= 'val_wer',
   min_delta=0.05,
   patience=3,
   verbose=True,
   mode='min'
)

trainer = pl.Trainer(max_epochs=2,
                     accumulate_grad_batches=1,
                     checkpoint_callback=False,
                     logger=False,
                     log_every_n_steps=5,
                     check_val_every_n_epoch=1,               
                     callbacks=[early_stop_callback],
                     gpus=0, 
                     # accelerator='ddp',
                     # plugins='ddp_sharded'
                    )

asr_model.set_trainer(trainer)

trainer.fit(asr_model)
```

**Environment overview**

Environment location: Google Colab
Method of NeMo install: !pip install nemo_toolkit[asr]
NeMo branch version: main
Learning Rate: 1e-3

**Environment details**

OS version : "Ubuntu20.04.3 LTS"
PyTorch version : "1.10.0"

**Comment by titu1994** [comment]
You are trying to force a character based model config onto a pretrained model which is subword based - this will not work. You have to use a subword model if you are finetuning of loading the weights of a subword checkpoint in such an automated way, of manually load the weight of just the encoder and nothing else.

**Comment by VahidooX** [comment]
Does it work?
Use the speech_to_text_ctc.py script while adding the following to your config:
  init_from_nemo_model:
      model0:
          path: pretrained_nemo_file.nemo
          exclude:["encoder"]

**Comment by MelikaBahmanabadi** [comment]
> You are trying to force a character based model config onto a pretrained model which is subword based - this will not work. You have to use a subword model if you are finetuning of loading the weights of a subword checkpoint in such an automated way, of manually load the weight of just the encoder and nothing else.

Thanks. Now I use `conformer_ctc_char.yaml` with `nemo_asr.models.EncDecCTCModel.from_pretrained(model_name="stt_en_conformer_ctc_small")` . But I have error `Model stt_en_conformer_ctc_small was not found.` which kind of conformer models can be used with char_based?

**Comment by titu1994** [comment]
Most of our pretrained models are subword based, especially Conformer and Citrinet, so there's no Checkpoint that's been pretrained for character based encoding.

**Comment by MelikaBahmanabadi** [comment]
> Most of our pretrained models are subword based, especially Conformer and Citrinet, so there's no Checkpoint that's been pretrained for character based encoding.

OK, Thank you. I test again `asr_model = nemo_asr.models.EncDecCTCModelBPE.from_pretrained(model_name="stt_en_conformer_ctc_large_ls")` with `conformer_ctc_bpe.yaml` but in set up optimization I have error: `Interpolation key 'model.sample_rate' not found`. Would you pleasehelp me what should I do?

**Comment by titu1994** [comment]
Please follow the tutorials for CTC finetuning on another language (the same technique applies if language remains same), that has the correct way for you to fine-tune your model. When finetuning you will most of the time not need the base config called comformer_ctc_bpe.yaml

**Comment by VahidooX** [comment]
This error happens usually when you take a subpart of the config out. The problem is that hydra can not extract the value. To fix that, you need to set the sample_rate to 16000 in the config file or set in the config object before passing it out to another method.

**Comment by MelikaBahmanabadi** [comment]
Hi, Would you please share with me a collab notebook fire for fine-tuning conformer_ctc_bpe? 
Thanks

**Comment by VahidooX** [comment]
You may check here: 

https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/ASR_CTC_Language_Finetuning.ipynb

**Comment by MelikaBahmanabadi** [comment]
> You may check here:
> 
> https://github.com/NVIDIA/NeMo/blob/main/tutorials/asr/ASR_CTC_Language_Finetuning.ipynb

Thanks but this solution is ok just for Citrinet model, not Conformer even for the Japanese language that by default is taught.
